update to latest & contact

whismerhill opened the pull request:
Hi @kingmooselord ,
made a PR to keep up your base code to endless sky master at the time of writing.

Mostly a pretext to get to contact you about your work.
I want to try compiling & running endless sky in VS2017 as part of trying to fix issues.
I have already compiled & ran endless sky in CodeBlocks before.

however compiling in VS2017 didn't work.
EDIT: managed to compile successfully
main change was to edit the project file to use *.h & *.cpp files which then includes all files in folders


Also this project needs "windows universal CRT SDK" component if compiling not in Win10 SDK (otherwise you get "Error	C1083	Cannot open include file: 'stddef.h': No such file or directory" & similar others)
Thanks for your attention. you can find me on discord in the endless sky community if you want : https://discord.gg/RtBkT4C

Elyssaen committed:
require <outfit> can require a specific number, or specifically zero

tehhowch committed:
Simple logic to determine if missile boat AI will reverse or turn around

Use reverse thrust if its capability exceeds:
 - the target's ability to close the distance
 - the current closing rate of the target
 - our forward thrust capability

Otherwise, turn around to use the better thrusters.

Elyssaen committed:
Simplifying NPC::HasFailed by cutting mustLiveFor

tehhowch committed:
Dialogs can respond to Esc or Ctrl+W

 - If Esc is not the "Show Main Menu" key, it will close the dialog
 - Closed dialogs
  - Choose "Cancel" or "Decline", if possible
  - Choose "OK" for information-only dialogs.
 - If Esc is the "Show Main Menu" key, that functionality has priority (e.g. the player can save when being offered a mission dialog).

tehhowch committed:
NPC "accompany" requires failIf value

 - With no 'failIf' value, a mission with NPCs that are listed as "accompany" will not fail if it becomes impossible to complete (e.g. one of the ships is destroyed)
 - With a 'failIf' value, the player is immediately shown the "Mission failed" message and the mission will be removed upon landing.

flaviojs committed:
Fix not being able to take back orders that require a target

To take back an order like "gather" or "focus fire" you issue it again.

When one of your ships is unparked in another system it will first
travel to your system. While it is traveling, orders that require a
target could be given but not taken back.

Commit:
If a LocationFilter is called on a Planet, compare the filter's government list to the government of the planet, not the government of the system.

tehhowch committed:
Support 'visit' & 'visit planet' in events

 - Adds ability for a game event to give the player knowledge of a system or planet.
 - Visit actions occur after any 'unvisit' actions in the same event
 - Update the structure of GameEvent::Load

tehhowch committed:
Fix ships selected by number keys in Shipyard/Outfitter

 - Ensure only ships which can appear in the Shipyard/Outfitter are selected.
 - If the currently selected ship is not in the new selection of ships, update it.

tehhowch committed:
Standardize engine flare sprite location (#2506)

* Standardized engine sprites

All engine flares now reside in subfolders

* Remove redundant names

If moving to subfolder system, no need to say "coalition flare/coalition large flare" when we already know it is the coalition flare. "coalition flare/large"
This matches the characterization of atomic sprites.

* Revert graviton change (placeholder tech)

flaviojs committed:
Keep partially collected flotsam (#2955)

* Keep partially collected flotsam

The leftover units would vanish when only part of the flotsam fits the
cargo hold. This means that you can lose cargo just by transfering it
to another ship of your fleet (jettison cargo, then guide other ships to
the flotsam).

This commit changes that behaviour, only removing the flotsam when there
are no more units. Only 1 ship can collect from the flotsam on each step.

* Jettison partially collected flotsam from the collector ship

Place the flotsam as if it was collected, partially stripped,
and then thrown back out.

* Replace placement of partial flotsam with an adjustment in velocity.

Flotsam transfer logic is moved to Flotsam::TransferTo and the flotsam
velocity is stabilized in proportion to the amount being transferred.

* Cast to double with static_cast.

Hadron1776 committed:
Allow modders to override default crew attack/defense values per government (#2708)

* Allow modders to override default crew attack/defense values per government (#40)

* Resolve a type mismatch (#41)

* Move government definition (#42)

* Resolve some style issues (#43)

* Do not allow attack/defense values to be negative (#46)

Commit:
Moving the check for negative crew attack and defense to where the values are loaded, since they're loaded once but queried many times.

Commit:
Made it possible to "add" or "remove" personalities from a fleet (based on @Hadron1776's PR).

Commit:
Fixing file list alphabetization in the CodeBlocks project.

Hadron1776 committed:
Add cloaking heat attribute (#2844)

* Add cloaking heat attribute (#116)

* Add cloaking heat tooltip (#117)

* Cloaking should now be possible :P (#120)

* Don't decloak just because heat is high.

Otherwise, even a cloaking device with no "cloaking heat" would not work on a ship with 90% heat. Plus, if you overheat while cloaked, it's your own darn fault.

tehhowch committed:
Fix NPC HasSucceeded() to use only recorded ShipEvent information (#2859)

* Fix NPC HasSucceeded() to use only recorded ShipEvent information

NPC::Do calculates whether a given ShipEvent causes success or failure by comparing the value of HasSucceeded before and after recording the ShipEvent.
 - Previous implementation inherited knowledge about the ship's Disabled status from the ship itself, and not from the recorded information.
 This meant that `evade` missions could never determine that the incoming ShipEvent::DISABLE was the one that allowed for success criteria.
 - Learning of a new ShipEvent::ASSIST will appropriately remove a ship's known ShipEvent::DISABLE flag.

* Derelict ships require a recorded assist event

tehhowch committed:
Update allowed travel plan creation conditions (#2938)

* Update allowed travel plan creation conditions

 - Remove incorrect MapPanel error message when the player's flagship has no hyperdrive/jump drive installed.
 - Wormhole-only travel plan creation is possible for the player when no hyperdrive/jump drive is installed.

* HasSeen->HasVisited, to prevent incorrect errors

* Alls ship without hyperspace ability can plot wormhole-only travel plans

Rather than check that DistanceMap was initialized for the player, check
if the calling ship's system has a wormhole available.

* Create strings only once

AMDmi3 committed:
Honor environement (#2949)

* Honor environement

Pass environment through SCons. For instance, this is required to pass CCACHE_* variabled to ccache.

* If carrying over all environment variables, no need to handle any individually.

Commit:
For flight check tooltips, check for mouse hover right when drawing the flight check icon so there's no need to scan through all the click zones later.

jjhankins committed:
Add a combat rating check to "Hunted by State" missions. (#3122)

* Add a combat rating check to "Hunted by State" missions.

* A more reasonable CR for the Hunted by State missions.

Commit:
Tweaked the "orbits" UI widget so there's space for a much longer system or planet name.

Commit:
Added colored "faction markers" to the targeting display.

tehhowch committed:
Mission NPCs ignore mining time/count limits

warp-core committed:
Typo fixes

tehhowch committed:
Extend reference-checking in GameData::CheckReferences (#3116)

* Extend reference-checking in GameData::CheckReferences

 - Add Name() to Minable & Phrase
 - Add IsDefined() to GameEvent

* Enable reference-checking for GameEvents and Missions

* IsDefined->Name and other feedback

tehhowch committed:
Ship Variant definitions can add attributes without needing to fully specify all other base attributes (#2959)

* Ship variants can add individual attributes

via the "add attributes" node. All base attributes can still be overridden by specifying the "attributes" node.

* Store "add attributes" values in baseAttributes

Ship::Save() only writes out baseAttributes.Attributes(), i.e. the values from "add attributes" were being lost when loading instantiated missions from a save file.

* NPCs created from a partial node description use the desired name

* Use same "add" key/value syntax as other classes

 - The only key currently supported is "attributes"

* Revert key-value-valueIndex changes

* Use supported ship specification method in Syndicate Target Practice

* NPC::Load prints warning, skips invalid use of ship specification

 - individual ships should not be fully specified in a mission's NPC definition - only `ship "model/variant" "name"`, or use of custom fleets.

* Fix typo and touch up warning message

flaviojs committed:
Use selected ship/outfit as scroll reference. (#3044)

* Use selected item as reference in ShopPanel.

In the outfitter, when you have an outfit selected and choose a different
ship it will keep the outfit in the same screen height.
This is needed because when a new ship is selected it can add or remove
outfits before the selected outfit, causing the selected outfit to be
placed in a different position.

It is also able to auto-scroll down and tries to make the image visible.

* Deselect selected item when you click on other areas of the main panel.

This allows you to cancel auto-scroll by clicking somewhere else in the
main panel.

* Clarify main top and bottom coordinates.

Commit:
Fixed the outfitter's auto-scroll when toggling outfits for sale or in cargo via the "key" buttons. Aslo reverted the behavior for click-and-drag scrolling so it won't clear the current selection.

flaviojs committed:
Clarify "Landing on:" navigation mode. (#3112)

* Clarify "Landing on:" navigation mode.

It can display "Landing on", "Can land", and "Cannot land".
It adds "(mission)" when it is the destination of a mission.

* Remove "(mission)" from "Landing on" navigation mode.

* "Can land on: Planet" makes more sense than "Can land: planet."

jafdy committed:
Add missing " mark

tehhowch committed:
, -> .

Somebody oops'd :)

Commit:
Refactored Engine::CalculateStep(), splitting each set of calculations into a separate function to improve clarity and to open a path for possibly multi-threading the most expensive calculations.

Commit:
Switched to using a vector for Projectiles and Effects, to reduce memory allocations and improve cache performance.

Commit:
Added a new "Visual" class that represents just the information we need to track for an Effect once it's instantiated. Using it in place of Effect saves a string copy and a good deal of memory for each active visual effect.

Commit:
Typo fix.

Commit:
Added a "Dictionary" class that works like an STL map<string, double> but allows faster lookup and no string allocations.

Amazinite committed:
Capitalized "Paradise Planets" in Follower's description (#3135)

* Capitalized "Paradise Planets" in Follower's description

* ; -> :

Commit:
Make sure a ship's destruction generates an event even if it self-destructed.

Commit:
Since CollisionSet is only used with Ships right now, switched it to using Ship pointers instead of generic Body pointers.

Commit:
Switching to just using a Weapon pointer in places where that's all that is needed instead of a full Outfit pointer.

tehhowch committed:
Prevent matching LocationFilters to planets that have no system

 For planets which are only fully defined after certain GameEvents, it is possible that the game attempts to obtain the government of their system. Since these planets are not in a system yet, this would generate a segfault due to an unhandled nullptr use.

tehhowch committed:
Add "Pug Wormhole" planet definition (#3138)

* Do not warn for wormholes that are finished via GameEvent

 - Wormhole "planets" are given a system, but unless also defined with a `planet` entry, would not pass GameData::CheckReferences until the wormhole was linked to another system

* Add empty planet definition for the Pug Wormhole

 - Revert change from previous commit that extended GameData::CheckReferences for planets to check for a lack of system too.

Commit:
Made it so that wormhole links are only drawn if the planet has a description. (This makes it possible to create wormholes that NPCs land on but that don't have links drawn.)

tehhowch committed:
Do not draw the flagship if it is destroyed (#3142)

* Do not draw the flagship if it is destroyed

 - Only send ShipEvent::DESTROY when the ship is destroyed (e.g. landing and forget removals do not generated events).
 - Use cached variables where sensible or already defined.

* Merging with my latest changes.

Commit:
Typo fix.

Commit:
Also adding a check for planets with no system to Planet::GetGovernment(), just for the sake of defensive programming.

tehhowch committed:
Fix Outfitter Attribute display bug (#3150)

* Update OutfitInfoDisplay

 - Change `auto` to the type `pair<string, double>` to avoid adding char
 - push_back to emplace_back
 - Simplify display of per-second values
 - Omit per-second values if reload = 60 (i.e. per second = per shot)
 - Separate non per shot / per second values from values which should be printed as per shot / per second
    - Addition of new damage types no longer requires updating the `isContinuous` offset

* Keep printing per-second values for weapons which fire once per second

* Iterate over the natural type pair<const char *, double>

tehhowch committed:
Doombat self-destruct

Commit:
Updated the faction icons to be a bit less tacky-looking.

Commit:
Switched back to using generic Body in CollisionSet, rather than Ship.

Commit:
Added a new CollisionSet function that checks collisions along a line defined by a start and end instead of by a Projectile object.

Commit:
Switched to using CollisionSet for asteroids and minables, and cleaned up some of the code related to them.

Commit:
Took out AsteroidField::Asteroid::Collide(), which is no longer used.

Commit:
For the faction icons, make sure they're always aligned with a pixel so they aren't blurry.

tehhowch committed:
Allow using fuel damage (fix from @Nechochwen-D)

tehhowch committed:
Fighters that are launched should not be marked for removal

flaviojs committed:
Only rehire extra crew when the player has a flagship.

Fixes crash when you create a new player with the preference
"Rehire extra crew when lost" enabled.

tehhowch committed:
Remove deprecated bribe action

 - Check was added in 03a9a089df3e9440e784604de0c80e8365a447be, when the fine action did not check if the player was freshly loaded.
 - Check was deprecated by 2866bcf1c65abd90db07ffb7e470c2f73bf9485f

Amazinite committed:
Fixed improperly indented fleet ship

flaviojs committed:
Suspend the boarding approach when too far away or too fast.

The boarding approach changes the ship angle, velocity, and position
artificially (no engines). It is activated when you are close enough
and slow enough.

After activating it, you can become too far away by selecting another
target and you can become too fast by getting hit. With this commit
the boarding approach will be temporarily suspended until you get
close enough and slow enough again.

ph2000bis committed:
Modified description of Sek Alarfarat (#3209)

* Modified description of Sek Alarfarat

Appart from fixing a mistake, the description misleadingly lead to
believe that the mantle of a planet is liquid, which is not the case.

* rechanged to a more simple explanation

Pointedstick committed:
More ship names (#3221)

* More words for the civilian and pirate ship names, and higher frequency
for the standalone list of civilian ship names

* Messed up the syntax the first time; fixed now

* @DrBlight insists that there must be ships with "Teacup" in the name

Commit:
Added new "landscape" images from @sivael for human space station interiors.

Amazinite committed:
Expanded the descriptions of the 4 main Hai ships (#3215)

* Expanded the descriptions of the 4 main Hai ships

* Feedback changes

* Feedback changes

* Feedback changes

Commit:
Adjusted the layout of the "compare" boxes in the shipyards and outfitters map panels to work better on wide, short monitors.

Janaszar committed:
Fix misspelling of deceleration (#3222)

* Fix misspelling of deceleration

* Update Ship.cpp

Commit:
If you quit while in the new pilot conversation, a blank pilot is no longer created.

tehhowch committed:
Add 'notoriety' tracking, for tracking ShipEvents vs. entire factions

 - If a hostile ship boards a member of a faction, that faction is more likely to target that ship than a hostile ship that has not boarded their ally

Commit:
Made it so pirates won't steal passengers when plundering. Also cleaned up and simplified the CargoHold transfer code.

Commit:
Since passengers can no longer be "plundered," there's no need to check for that when a player's ship is boarded.

Commit:
Made the saved game date tooltips only appear if you hover over the same item for a full second.

Commit:
An error is now shown if a mission has no name or if the same mission is "loaded" from two different definitions.

Commit:
Made it possible to anchor different elements in an Interface to different corners or edges of the screen.

Commit:
Combined the heads-up-display into a single "hud" interface specification.

Commit:
Made it possible to customize the position and size of the ammo, escorts, and message lists in the heads-up display.

Commit:
Cleaning up a leftover debugging #include.

Commit:
Since special "persons" always enter via hyperlinks, don't spawn them in systems with no links.

Commit:
Added support for special "persons" with multiple ships. The "person" can regenerate all its ships unless you kill the flagship.

Commit:
Made it possible for the start conditions to include the player owning one or more ships.

Commit:
Reduced the lifetime for the particle beam impact sprite, because the animation is over after 15 frames.

Commit:
Made it so illegality and stealth is an attribute of the mission as a whole (applying to both cargo and passengers) rather than just an attribute of the cargo.

tehhowch committed:
Ctrl-Shift-Arrow key moves selected ships on PlayerInfoPanel (#3229)

* Ctrl+Shift+ Up/Down arrow key moves selected ships on PlayerInfoPanel

* Use shift direction to allow dragging to the end position

Commit:
Making sure no invalid iterators are created in PlayerInfo::ReorderShips(), because OS X may complain about that even if they are never dereferenced.

Commit:
Potential error fixes from @AskePit.

tehhowch committed:
Remove use of static inside anonymous namespace

tehhowch committed:
Create player conditions for mortgages, fines, back pay, and credit score (#3164)

* Add summary methods to Account and update Account-based autoconditions

 - Return a count of mortgages and fines in the account.
 - Return the total owed, broken into mortgages and fines.

Update account-based autoconditions

 New conditions for
  - # of active mortgages
  - # of active fines
  - value of active mortgages
  - value of active fines
  - value of unpaid salaries
  - player's credit score

* Remove autoconditions of the number of fines/mortgages
 - The total amount is a much more useful metric than the number of each

Commit:
Switched to a more readable way of querying the player's debt amounts.

Commit:
Added an "asteroid scanner" outfit. Also made it possible for turrets to track minable asteroids and for target "crosshairs" to have something other than 4 points.

Commit:
Made solar collection and ramscoop rates vary from system to system depending on the star(s).

Commit:
Added the file that specifies the star attributes.

Commit:
Added outfit images for the surveillance pod and asteroid scanner.

Commit:
Fixed a bug where the asteroid target color would be whatever color was used for the previous ship.

Commit:
Modified the outline shader to show some detail inside the ship, instead of just the silhouette.

comnom committed:
edit marauder manta's armament to match its outfits

Commit:
Added a tooltip for the asteroid scanner.

Commit:
Using the new fancy "outline" rendering in the ship info panel, too.

Commit:
Avoid pixelation in the ship info panel by never drawing the ship outline at more than 100% zoom.

Commit:
Fixed a situation where, if a hardpoint specified a weapon outfit that the ship didn't have and that outfit hadn't been loaded yet, it wouldn't be properly removed.

Commit:
Added a "tactical scanner" that shows additional information about your target.

Commit:
Your escorts in the "Syndicate Capture" mission will now help you fight the Syndicate ships.

Commit:
Split the Deep fleets into a Merchant component and a Deep Security component.

Commit:
Added some unique hails for the Deep Security forces.

gunqqer committed:
Fix a bug where some source files are not included in the XCode Project for OS X (#3238)

* Fix a bug where some source files are not included

On OSX the build fails because some source files aren't included in the XCode project, this should fix it but I honestly don't really know what I'm doing

* Update project.pbxproj

* alphabetizing stuff

* Removed extranius changes

These changes are probably due to me poking around things in XCode, if they need to be changed I don't know, but this PR isn't about it

* Removed extranius changes

These changes are probably due to me poking around things in XCode, if they need to be changed I don't know, but this PR isn't about it

* Removed random character

tehhowch committed:
Surveillance Pod has some tactical scan power

tehhowch committed:
Overheated/Hyperspacing ships don't need to decide on actions (#3247)

* Overheated/Hyperspacing ships don't need to decide on actions
 - Hyperspacing ships can still (de)cloak and autofire weapons, but have no ability to change their movement strategies.
 - Overheated ships are effectively disabled, but IsDisabled explicitly does not return true for a ship which is solely overheated (so the radar can blink their pointer properly)
 - Stranded/Disabled ships shouldn't ask overheated or hyperspacing ships for help since that help may be a long time coming.

* Overheated but in need of help asks for help

* Feedback

tehhowch committed:
Missing <algorithm> include

Commit:
Added a new weapon for the Tele'ekite Wanderers, the Moonbeam.

Commit:
Boosted the damage and energy usage of the Moonbeam to better differentiate it from the Electron Beam (which is supposed to be at least half a tier lower, technologically).

Kryes-Omega committed:
tactical scanner image (#3250)

* uploaded tactical scanner image

* changed tactical scanner image

* fixed post-processing for image

* fixed lighting for scanner

* Update copyright

Nescio0 committed:
Update weapons.txt

Commit:
Cleaning up the code for the outline shader.

Commit:
Added support in ImageBuffer for loading multiple frames of an animation into one buffer.

Commit:
Added an ImageSet class, which will store all data that must be loaded for a given sprite and all functions for parsing sprite paths.

Commit:
Added functions for bulk-uploading sprites. (But, they're not in use yet.)

Commit:
Switched to using the new ImageSet class for batch-loading of sprites. (But still with each frame going into separate textures.)

Commit:
Switched to using 2D "texture arrays" for sprites, so all the frames fit into a single texture.

Commit:
Batching the draw commands for projectiles and effects.

Commit:
Removing some unnecessary std:: prefixes.

Amazinite committed:
Fixed a phrase typo

Pointedstick committed:
Reduce Bastion and Splinter minimum crew complements to make them more
attractive options

luiges90 committed:
Coalition "Tourists Home" appear to have attribute applied wrong way

Commit:
Propagate the "WAIT" command down to escorts of escorts, to synchronize jumping.

Commit:
The player will now remember minable types they have targeted in a system even if they didn't harvest anything from them.

Pointedstick committed:
Even more ship names! (#3262)

* Even more ship names!

* A few last-minute additions

* Correcting last commit

Commit:
Reducing the frequency of cardinal directions in "...of the X" ship names, because there's only four directions vs. dozens of generic nouns.

Pointedstick committed:
More ship names, and in particular more male and female names

Commit:
Automatic aiming now works with asteroids if you have an asteroid scanner.

Commit:
Don't highlight minables while jumping.

Commit:
Tweaking the Moonbeam to make it look less like the Heliarch beam weapons.

Commit:
Added two new Wanderer ships.

Commit:
Added a tooltip for tactical scan power.

tehhowch committed:
AI does not autofire weapons with firing force if preparing to jump

 - Switched Outfit * to Weapon * and updated variable names
 - Updated the includes to reflect the used classes.

toilethinges committed:
Missing quotation mark

tehhowch committed:
Consider both trigger and blast radius when firing all weapon types

 - Previous behaviors did not consider blast radius for non-homing weapons (such as Heavy Rockets), or trigger radius at all.

tehhowch committed:
Ships with non-safe blast radius weaponry do not follow the projectile

 Instead, they attempt to keep the minimum safe distance (blast radius + trigger radius) between them and their target ship.
 - Behavior change at 125% M.S.D.
 - Included consideration of relative velocity

Commit:
Typo fix.

flaviojs committed:
Fix binding of sampler tex.

For samplers, Mesa3D expects the argument to be int.
It was being sent as uint.

Amazinite committed:
Slightly buffed the Proton Gun in order for it to compete with the range and hit force of the Particle Cannon and the heat damage of the Plasma Cannon

Commit:
Simplified the OutlineShader, and modified how tweening is done so it will look better with animated sprites.

Amazinite committed:
Prevent Unfettered Aid jobs from having a destination in occupied Wanderer systems

Alkallid committed:
Add period

Alkallid committed:
Tense fix

tehhowch committed:
Add text substitution for <waypoints> in missions

Commit:
Made it so your behavior toward other factions won't affect the Pug's attitude toward you.

tehhowch committed:
Wrap messages when drawing them to prevent overflow (#3313)

* Wrap messages when drawing them to prevent overflow

* Increase padding on right edge to avoid overlap of ammo icon

tehhowch committed:
Use <waypoints> substitution (and quote a few user-customizable tokens)

Commit:
Modified the "ammo" HUD element to center the ammo icon boxes instead of left-aligning them within the given rectangle.

Commit:
Making the proton fragment blur and fade out a bit more at the end of its range.

mootootwo committed:
enable "start of hostilities" to trigger Republic rep penalty (#3322)

* add reputation penalty to "start of hostilities"

moves reputation penalty from mission into event, allows player to become hostile mid-mission.

* rep penalty for "Defend Sabik" moved to event

* start hostility timer increased by 1 day

the mission deadline allows for completing the mission by the end of the final day, but the event was triggering at the beginning of that day.  

Moved the event timer out one day.

* reputation change should take minimum

changed per tehhowch.

* made republic reputation penalty a 'minimum'

changed = to <?= per tehhowch

tehhowch committed:
Swapped two mission NPCs to their proper government

 - Due to the temporary ceasefire across all FW-Republic conflict, these escorts can properly belong to the Free Worlds government rather than the player's government.

mootootwo committed:
converted reputation adjustments to minimums

per @tehhowch 's comments on #3322

Pointedstick committed:
Proton Turret (#3316)

* Proton Turret

* Corrected mass to match space taken up

* Made sure that the new (Proton) variants appear in alphabetical order

* Review feedback

* Reduced to 34 tons (giving it effectively a 6-ton base and mount), which
should hopefully make it a bit more attractive

* Added copyright info

* Updated hardpopint image (drop shadow, correct lighting)

Commit:
Made the target pointer also show up for target asteroids.

Commit:
Typo fix.

Commit:
Fixing a variable that should be const in the outline shader.

ph2000bis committed:
Wanderer hail diversification (#3284)

* Wanderer hail diversification

Added a bit of diversity to the wanderer hails, that could otherwise get
very repetitive late-game

* fixed inconsistency

* Syntax fixes

* More fixes, and additions

added variety, and removed a wrong comma on one of the possibilities

* Addtions

* Fixed a gramar issue

Pointedstick committed:
Fixes for the Proton Turret

gunqqer committed:
Update file references in xcodeproj

Adds BatchDrawList, BatchShader, and ImageSet (.cpp and .h) to
EndlessSky.xcodeproj/project.pbxproj , fixes linker problems when
building from source

tehhowch committed:
Added a government specifier to a set of ships that did not have one

 - Rather than let these NPCs use the player government, set them to be "Free Worlds" and added the `escort` personality tag so they are still shown as your "convoy."

Commit:
Typo fix. (@themightygrunt)

Commit:
Renamed "attributes" to "in" variables for compatibility with more recent OpenGL versions.

tehhowch committed:
Allow non-player ships to emit ShipEvent::ASSIST when repairing ships

 - Also adds ShipEvent::BOARD and ShipEvent::ASSIST to the list of ShipEvents that only count toward NPC completion when performed by the player

tehhowch committed:
Change Outfit * to Weapon *, warn on incorrect armament specifications

 - Swapped the Outfit pointers to Weapon pointers for the associated ship firing methods
 - Print warning messages for incorrect armament specifications:
   - guns positioned in turret hardpoints
   - turrets positioned in gun hardpoints
   - non-weapons positioned in any hardpoint.

tehhowch committed:
Carried ships forget their target system/object when reboarding

 - Directly clear the carried ship's target system and target StellarObject upon boarding its parent.
 - Add a nullptr check for `currentSystem` in Ship::IsReadyToJump.
 - Add a nullptr check for `currentSystem` in Ship::JumpFuel

Alkallid committed:
Tense Fix

Commit:
Fixing a bug where FinishLoading() is called more than once if an NPC ship is copied from a stock ship.

Commit:
Don't show a "profit" relative to the cost basis for a commodity that you have in the holds of absent ships but do not have any of on-planet.

Commit:
Adding back in one defensive check and fixing some formatting.

tehhowch committed:
Avoid creating a passenger reference when 0 passengers are transferred (#3367)

* Outfit, Commodity, and Passenger transfers require a non-zero amount

 - Otherwise a reference to the item is created, even though the quantity is 0.
   - This reference would cause mission failure for destroyed ships even if they were not carrying mission cargo/passengers.

* Sanity check for non-zero passenger count

tehhowch committed:
Revert automatic assertion in FinishLoading that a ship is not disabled

 - 2633668c9f9c5933cbf8f0a98c855ab7b9345d20 made it so that every ship that FinishLoading is called for considers itself not disabled - even if its current hull and shields value mean it should be disabled.

Commit:
Modifying BatchShader to not rely on OpenGL being able to use pointers to vertex attributes in main memory.

tehhowch committed:
Certain ships should not auto-assist their disabled parent (#3370)

* Carried ships should not auto-assist their disabled parent

* Derelict ships should not auto-repair their parents either

tehhowch committed:
Implement a ShipEvent queue to allow delayed ShipEvent handling (#3371)

* Implement a ShipEvent queue to allow delayed ShipEvent handling

 - Handling ShipEvents can result in new UI elements.
 - Only one UI element should be created and displayed to the player at a time (i.e. do not display a dialog and the BoardingPanel simultaneously).
 - Callbacks from one UI element from a given ShipEvent may result in status changes that should influence other UI elements created from the same ShipEvent (e.g. if a mission conversation offered upon boarding an NPC results in that NPC's destruction, the player should not be able to plunder said NPC).
 - Fixes a preexisting issue where the boarding panel would be offered if the player dies in a conversation.

* Switch from GetUI()->IsTop(this) to GetUI()->Top().get() == this

 - IsTop() checks only drawn panels, not panels that are requested but not yet drawn.

Commit:
Fixed an improper weapon category.

Commit:
Whether a function parameter that is passed by value is const or not is an implementation detail, not something that needs to be defined in the class interface.

Commit:
Boarding mission cargo should go into the player's flagship, not the PlayerInfo cargo hold.

tehhowch committed:
Refactor MapPanel's Draw___ methods (#3379)

* Refactor MapPanel's Draw___ methods

 - Improve mod coloring support via expanded use of interfaces.txt:
   - Switch the minimap from a hardcoded position to an Interface
   - Use travel plan colors from interfaces.txt
   - Use System name, link colors from interfaces.txt
 - Move the increment of `step` from MissionPanel to the base MapPanel
 - Add comments to DrawMiniMap, DrawTravelPlan, and DrawWormholes
 - Used and updated MapPanel constants
 - Draw the Travel Plan after wormholes, since wormholes by default have non-zero alpha.
   - This removes the small, dark purple centerline from drawn travel plan routes that use wormholes
 - Adds function to Planet class to export all systems in which it can be found
 - Do not show the "planet description" box if the planet in question is a wormhole.
 - Change DrawWormholes iteration scheme from "every StellarObject in every System" to "check only planets"
   - Fixes bug in DrawWormholes caused when two wormholes reference the same system (i.e. Cardea), which cause the wormhole link to be drawn over a previous directional arrow

* Move the mini-map position specifier into the "hud" interface

Commit:
Simplifying the wormhole-drawing code.

Commit:
Moving the mini-map back up where it was, to be less intrusive on small screens (e.g. 1366x768).

Lorantine committed:
typo fix for @UnciaPrima (#3399)

Turtleroku committed:
Typo?

cut "out" territory in half

tehhowch committed:
Inscrutable targets are impervious to tactical scans too

tehhowch committed:
Enable any mission conversation to require the player to launch

 - Adds a "BasicCallback" to enable conversations that come from non "on offer" conversations to force the player to depart the planet.
 - Guards setting "shouldLaunch" to true while in flight, as a boarding/assisting mission could otherwise set this and cause odd behavior on the next landing.

tehhowch committed:
Update "Grudge" messaging

 - Use an anonymous function to determine if the given ship is capable of sending a hail (e.g not mute or out-of-system or uncrewed)
 - Unify "author" formatting for message printing

Brick63 committed:
Add a new korath Hand to Hand weapon (#3321)

* added the korath rifle

* changed the number each ship carries

* made the Description better

* number change

* adjusted the numbers

* feedback changes

* number changes to keep defense

* more fixes

* Feedback changes

removed the illegality and changed the description to fit

* Fixed the Description

* Update korath outfits.txt

* Update copyright

* fix

* fix

tehhowch committed:
Refactor PlayerInfo::Land (#3208)

* Refactor on-land mission checks and fining into own functions

 - Mission updates based on landing are now in StepMissions()
 - Planets for which mission clearance has been given (or if infiltrating) will not fine the player

* Track visited waypoints & stopovers

 - This enables remembering that a mission gave clearance to a particular planet
 - This enables remembering which systems needed to be visited for a mission

* Remove deprecated government assignment in PlayerInfo::Load

 - ships.back()->SetGovernment was added to PlayerInfo::Load() in 0783cd49ea5204bab9f126723a61833d25c5118f
 - Ship::Load() sets the ship's government to be the player's government as of ef41ef02021bd43954ba25631922dae6c928a8a9
 - ApplyChanges() (called at the end of PlayerInfo::Load()) performs ship->SetGovernment for all the player's ships, in case the player's swizzle has been altered in an event

* Use a better format for serializing visited stopovers and waypoints

* Fix bug related to handling of the player's death

 - If killed due to an atrocity outfit while landed, the player's ship list needs to be cleared (otherwise the player can simply press 'D' and take off again).

Commit:
Fixed the last commit so dominated planets will not fine you (instead of dominated planets fining you even if you have mission clearance).

tehhowch committed:
Truncate long player / player flagship names, and plugin names (#3395)

* Truncate long player / player flagship names, and plugin names

* Restored proper RHS margin for plugin text

Commit:
Made the outline shader use the 2x image when available and appropriate.

Commit:
Tweaked the star field rendering to reduce the "sparkle" at low zoom levels.

Commit:
Increased blaster speed and range so it's useful in different situations than a laser. Also reduced the electron beam damage significantly, and the heavy laser by a tiny amount.

Pointedstick committed:
New Energy Blaster and Modified Blaster sounds (#3306)

* Replaced the energy blaster and modified blaster sounds with ones that sound
more like sci-fi space guns

* Clipped the sounds to 0.3 seconds, matching the prior length

tehhowch committed:
Define additional filled colors in interfaces.txt (#3425)

* Define additional filled colors in interfaces.txt

 - Further support for modified interfaces / reskinning.

* Added comments to color definitions

* Reflect name change in code too

Commit:
Increased the heat dissipation of the two non-combat starting ships to make it a bit easier for them to evade missiles.

Commit:
Reduced the turn rates of most turrets, making it so the heaviest ones have trouble tracking fast ships.

Commit:
Fixed a bug where the stream reload counter reset whenever a weapon of a given type was removed, causing all nuclear missiles to fire at once.

Fzzr committed:
Minor spelling fix (#3437)

* Remove extra "s" in "Coalitions"

* Use "with which" instead of "that... with"

* Revert "Use "with which" instead of "that... with""

This reverts commit 6f1c50eb6e63ea24fc776da5c6393b5eeb336e1a.

tehhowch committed:
Alphabetized variants.txt (#3439)

* Reorder Argosies

* Reorder Arrow

* Reordered Bactrians

* Reordered Barb

* Reordered Bastions

* Reorder Behemoths

* Reorder Berserker

* Reorder Bounder

* Reorder Bulk Freighters

* Reorder Carriers

* Reorder Clippers

* Reorder Corvettes

* Reorder Cruisers

* Reorder Dreadnought

* Reorder Falcons

* Reorder Firebirds

* Reorder Flivvers

* Reorder Freighters

* Reorder Frigate

* Reorder Furies

* Reorder Gunboat

* Reorder Haulers

* Reorder Hawks

* Reorder Headhunters

* Reorder Heavy Shuttle

* Reorder Lance

* Reorder Leviathans

* Reorder Mantas

* Reorder Mod. Argosies

* Reorder Mules

* Reorder Nest

* Reorder Ospreys

* Reorder Protectors

* Reorder Quicksilvers

* Reorder Rainmaker

* Reorder Ravens

* Reorder Roost + Scout

* Reorder Skein

* Reorder Sparrows + Star Barge

* Reorder Wasp

* Reorder Vanguard

* Reorder Splinters

tehhowch committed:
User-defined colors & deprecated function (#3431)

* Remove deprecated function declaration

 - Function was removed in https://github.com/endless-sky/endless-sky/commit/cdb1cb881d04b246feee03dff4ec0028f34f6327#diff-e0a59618d76b767cb62fa83a9f007c59L295

* Switch to colors in GameData, add some method comments

* Deprecate "close" and "far" colors for "medium" and "dim"

* Show a button for the Find dialog

* Revert "Show a button for the Find dialog"

This reverts commit 14901ad25707f8fd05251d54e45e053a7c3209bd.

tehhowch committed:
Update AI::DoScatter to ignore out-of-system ships

flaviojs committed:
Show licenses in outfitter (#3068)

* Show owned licenses in the outfitter.

The license will only show if there is an outfit with the correct name.
Player condition "license: NAME" is mapped to outfit "NAME License".

* Add outfit "Militia Carrier License".

* Add outfit "Coalition License" and a matching log entry.

* Add outfit "Wanderer Outfits License".

* Add outfit "Wanderer License".

* Add outfit "Wanderer Military License".

* Add outfit "Remnant License" and a matching log entry.

Commit:
Added some license images.

Commit:
Added images for the human licenses.

Pointedstick committed:
Standardize on the Appstream ID "io.github.EndlessSky.endless-sky"

Pointedstick committed:
Add <launchable> tag to associate AppStream data with Desktop file

warp-core committed:
Typo fix

Added `the` to `right next to back entrance that she pointed out.`
To give `right next to the back entrance that she pointed out.`
Also removed an unnecessary `go to` as the label it pointed to was immediately after it anyway.

Pointedstick committed:
Added <releases> tag to the AppStream file, including version number,
release data, and changelog for the latest release

warp-core committed:
Typo fix

warp-core committed:
Type fix

Changed: `...attacked you after you send out the data...` to `...attacked you after you sent out the data...`

tehhowch committed:
Add missing reputation serialization to conversation ConditionSets

 - Fixes issue where changes to the stored government reputations were not reflected in the current political landscape when the changes were performed via conversation `apply` node.

Pointedstick committed:
More ship names! (#3451)

* More ship names!

* Replaced the controversial name

* A couple more, because this is fun

Amazinite committed:
Typo fixes (#3471)

* Travelled -> Traveled

* Travelled -> Traveled

* Traveller -> Traveler

Commit:
Added a "news" system for displaying messages from local inhabitants you meet in the spaceport.

Fzzr committed:
Spell "grey" consistently in ship names per #3476

Commit:
Removed the iCCP profiles from the portrait images.

flaviojs committed:
When you click a system in the map mission view, iterate through all missions that involve the system. (#3162)

* Show all missions that involve the system.

This allows you to know which of the missions has the system as a
waypoint or has a system planet to land on.

* Comment and code style.

Commit:
Several related changes to fighters: they can now run their own systems when being carried (to generate energy, repair, etc.). Their heat level is now indepenedent of the ship carrying them. Fighters now retreat for repairs if badly damaged, and launch only when mostly repaired. And, repair priority is given to the least damaged fighter.

Commit:
Typo fix in the previous PR.

Commit:
The shop buttons are now highlighted if you hover over them (like all the other buttons in the game).

Commit:
In the shops, made it so when you hover the buy or sell button, the ships that the buy or sell will apply to are highlighted.

Commit:
Added support for system attributes, to be used when picking systems for waypoints, NPCs, or persons.

Pointedstick committed:
Adjusted the Meteor Missile's description  (#3270)

* Adjusted the Meteor's description to provide some more hints about how
best to use and defeat them

* Improved the description according to @endless-sky's comments

czartrak committed:
Added defense missions to the Core against Korath Attackers (#3325)

* Added defense missions to the Core against Korath Attackers

It seems logical that there should be core missions to defend against the Korath, so I wrote a few scaling from a few raiders, to battling World Ships.

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

* Update jobs.txt

Fixed indents and backticks

* Update jobs.txt

* Fix whitespace and alignment

* "Defenses" should be spelled "defense" when referring to a single battle

Commit:
Tweaked the mechanics of hit / firing force to increase its effect on large ships and decrease the effect on small ships, so large ships can't shrug it off completely and small ships aren't sent flying quite as far by every ship explosion. Also increased hit force for most missile weapons.

Fzzr committed:
Minor spelling and grammar corrections in Wanderers files (#3463)

* Use active voice for Winter Gale description

* Capitalize "Wanderers"

* Remove unnecessary comma

* Remove unnecessary hyphenation of "retrofitted"

* Revert "Use active voice for Winter Gale description"

This reverts commit dee8549ffc3440f8f5fead87c2b8af167ae1b2c0.

* Revert "Remove unnecessary comma"

This reverts commit cb6cfcedd8f3729c79d883a6a6201bc11dac8b86.

Pointedstick committed:
Fancier ships in paradise (#3386)

* Added some more fancy and passenger-oriented ships to the fleets that
hang around Paradise planets

* Corrected a formatting error

tehhowch committed:
Per request (#3416)

* Create "Power of the People", a modified Osprey warship

* Use common "friendly author" phrase

* Add Phrases

 - Amalgamation of various pop culture references, mostly
   - Han Solo
   - Ernest
   - Starship Troopers
   - The Fifth Element
   - Die Hard

* Update copyright file

 - Removed duplicated Vanguard entry

* Adding "friendly author" to Captain Nate (by request)

Hacklin committed:
Save a ship's base mass. (#3481)

* Save a ship's base mass.

* Manipulate the "mass" attribute the same way as other attributes.

* Use the chassis mass of the ship's model.

Fix broken save files of issue #3480

Commit:
Display an outfit's mass (if nonzero) right after the cost. Also, fixed the spacing of the attributes list for weapons with no non-weapon attributes.

Commit:
Fixed the "mass:" outfit info label and made the capacity lines and hardpoint lines group together.

tehhowch committed:
Use add attributes (#3483)

* Use "add attributes" to simplify Kestrel definition

* Use "add attributes" to simplify Marauder variant definitions

* Simplify "Archon (Cloaked)" definition

* Simplify "Deep River <N>" definitions

* Avoid variant definition using a variant as its base

 - Without dependency checking in FinishLoading, if a variant references another variant as its base model, there is no guarantee the base variant is fully defined yet

tehhowch committed:
Update Pirate planetary defense missions (#3482)

* Update pirate "planetary defense" missions

 - Fix incorrect opposition references in UI text
 - Add unpaid variants (25% chance) that are slightly easier than the paid variants
   - Same overall chance of triggering a defense mission

* Add reference checking of stock conversations

* Co-locate mission conversations and mission definitions

tehhowch committed:
Allow LocationFilter to specify attributes of a neighboring system (#3173)

* Allow LocationFilter to specify attributes of a neighboring system

 - Neighboring systems are those that are linked, and not those that are linked or within jump drive distance, to prevent matching to possibly inaccessible systems.

* Disallow in-line not & neighbor usage

* Reduce duplicated code

* Only check NeighborFilter matches when checking systems

* Perform simpler checks first

tehhowch committed:
Weapon blast strength decreases with distance from blast (#3276)

* Weapon blast strength decreases with distance from blast

 - To mitigate the damage falloff for weapons with a trigger radius, a multiplier based on the ratio of trigger and blast radius is used.
 - For a weapon with 0 trigger radius, the damage will range from 100% at the center, to 25% at the edge of the blast radius.

* Use worst-case distance estimate rather than exact computation

Commit:
Making torpedoes less effective against small targets (based partly on @Pointedstick's suggestions).

Commit:
Fixing "persons" whose death explosions won't work because they don't have a permanent base model ship.

tehhowch committed:
Player-owned ships are always "in-range" for the Tactical Scanner

tehhowch committed:
Asteroids should be "inscrutable"

Commit:
Fix divide-by-zero in the repair heat calculation.

tehhowch committed:
Mission "on enter" can define a LocationFilter to pick the system(s) (#3141)

* "on enter" actions can define a LocationFilter to pick the system

 - Example:
    on enter
      "system filter"
        government "Syndicate"
      dialog "You're now in Syndicate space!"

 - Without a system filter, any `on enter` actions that do not name a system will apply as soon as the player takes off (current behavior).
 - Each mission only supports a single `on enter` action without a named system (current behavior).

* Use a function to convert a distance filter into a near filter

  - This avoids needing to save the mission origin in the MissionAction
  - Use of both "near <system>" and "distance" filters in the same `on enter` action is not supported - only the "near" filter will be kept.
  - Fixed an extra indentation level with the saved LocationFilter.

* Forward compatibility for changes to LocationFilter class variables

* Support multiple "generic" on enter MissionActions instead of only one

 - "specific" on enter: explicitly identifies the system
 - "generic" on enter: uses a LocationFilter to match a set of systems (offering in the first one that matches), or has no filter and offers immediately on launch
 - All "specific" on enter actions are checked before any "generic" on enter actions for a given system
 - Only one on enter action will be triggered per jump.
 - Each "generic" on enter is checked in the order of definition in the mission datafile

* Update SetOrigin() usage

 - Reformat checks to avoid copying if nothing is to be done
 - Pass a system to MissionAction::Instantiate to guarantee a System * is available, even for assisting/boarding missions.

* Use didEnter to track both generic and specific "on enter" actions

Commit:
Cleaned up the "on enter" code a bit, and used it to improve the April Fools mission.

Commit:
Fixing a visual glitch in the Zenith planet images.

Commit:
Further fixes to the repair heat code. (@tehhowch)

Commit:
Added constructors that call Load() on a data element, so construction and Load() don't have to be done in two separate steps.

tehhowch committed:
Move PickSystem, PickPlanet from Mission to LocationFilter

 - Support choosing a random system or planet from a given LocationFilter outside of just mission instantiation

tehhowch commented:
Looks like your include directories aren't correctly set. 

The reason you're getting linking errors about BatchDrawList (and News) is because the VS2017 project file needs to specify them as a part of the project - if they aren't specified, then VS won't build them when it builds the project, and thus the linker can't find the functions that are being referenced from the files that are included in the project.

(For the record, a PR such as this one, while well-intentioned, isn't really very helpful. A better PR is one to the relevant branch that updates the VS project with the needed files.)

Commit:
More help for confused new players.

whismerhill committed:
added missing quotes on gascraft 

no idea if it's on purpose, while in doubt here's a PR.
it's also for better consistency with other entries imho.

Fzzr committed:
Add News.cpp and News.h to Xcode project (#3509)

* Add News.cpp and News.h to Xcode project

* Alphabetize source

Commit:
When displaying how attractive your fleet is to pirates, make sure it's between 0 and 100%.
